src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder.cs

```diff
@@ -379,6 +388,14 @@ private static string GenerateRuntimeSignature(IEnumerable<LockFileTargetLibrary
                 else if (export.IsProject())
                 {
                     referenceProjectInfo = GetProjectInfo(library);
+
+                    foreach (var dependencyReference in referenceProjectInfo.DependencyReferences)
```

Comment:
This allows the references to be listed as dependencies of the containing project, rather than as direct dependencies of the current project.